lib/base-compiler.js

```diff
@@ -856,9 +862,12 @@ class BaseCompiler {
     isCfgCompiler(compilerVersion) {
         return compilerVersion.includes("clang") ||
             compilerVersion.match(/^([\w-]*-)?g((\+\+)|(cc)|(dc))/g) !== null;
-
     }
 
+    isSelectionDAGCompiler(compilerVersion) {
+        return compilerVersion.includes("clang") || compilerVersion.includes("LLVM");
```

Comment:
We might want to make this an explicit tag on the compiler if it requires a patched compiler. We can help with that (plus the configuration). The "compilerVersion" trickery is mostly to support as many compilers as possible without changing config; but if the default will be "doesn't support DAG" with one or two that do, then we can consider the config approach.

Thoughts?

Reply:
Sounds like the right way to go.
LLVM not built with the patch will not support this.

By "explicit tag on the compiler" you mean... something in the config file?